MacOS binary problems.

I downloaded the latest version of jaguar for Macos (Monterrey) and I found some strange behaviors. To start the binary, when executing jag version, the execution does not end and it hangs waiting for something. In addition, it generates multiple stability problems in the operating system and I end up restarting it.
I have downloaded version 1.0.8 to test and I don't have that problem. The jag version command displays the information and finishes its execution appropriately. Although yes, when running a jag scan, it triggers an error since I don't have any device plugged in and after the error it hangs there.

This behavior is very strange and it is the first time it has happened to me in the years that I have been using MacOS.
I hope someone can reproduce this and we can find a solution.
I have also tried installing it using brew, but it reaches the make stage and hangs there.

Thank you very much for this tool, I am eager to try it.

> Could you also maybe try to build `jaguar` by hand:
> 
> ```
> mkdir debug-jaguar
> cd debug-jaguar
> git clone https://github.com/toitlang/jaguar
> cd jaguar
> mkdir build
> go build -o build/jag cmd/jag/main.go
> # See if it works.
> build/jag version
> ```
> 
> In theory the brew script does pretty much the same thing, but it has a few more Go environment variables set.


gaston@iMac Toit % mkdir debug-jaguar
gaston@iMac Toit % cd debug-jaguar 
gaston@iMac debug-jaguar % git clone https://github.com/toitlang/jaguar
Cloning into 'jaguar'...
remote: Enumerating objects: 3140, done.
remote: Counting objects: 100% (521/521), done.
remote: Compressing objects: 100% (185/185), done.
remote: Total 3140 (delta 273), reused 490 (delta 251), pack-reused 2619
Receiving objects: 100% (3140/3140), 851.80 KiB | 1.57 MiB/s, done.
Resolving deltas: 100% (1679/1679), done.
gaston@iMac debug-jaguar % cd jaguar 
gaston@iMac jaguar % ls
CODE_OF_CONDUCT.md	Makefile		docs			package.lock		tools
CONTRIBUTING.md		README.md		go.mod			package.yaml
LICENSE			cmd			go.sum			src
gaston@iMac jaguar % mkdir build
gaston@iMac jaguar % go build -o build/jag cmd/jag/main.go
gaston@iMac jaguar % cd build
gaston@iMac build % ls
jag
gaston@iMac build % ./jag version
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x16c604a]

goroutine 1 [running]:
github.com/toitlang/jaguar/cmd/jag/commands.JagCmd.VersionCmd.func3(0xc0001c8000?, {0x1ef8e40?, 0x4?, 0x184a5eb?})
	/Users/gaston/Toit/debug-jaguar/jaguar/cmd/jag/commands/version.go:30 +0x4a
github.com/spf13/cobra.(*Command).execute(0xc0001c8000, {0x1ef8e40, 0x0, 0x0})
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x863
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001b4300)
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(0x184bfb9?, {0x1970488?, 0xc00017d920?})
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:961 +0x47
main.main()
	/Users/gaston/Toit/debug-jaguar/jaguar/cmd/jag/main.go:34 +0x190
gaston@iMac build %

> build/jag version


gaston@iMac jaguar % git checkout floitsch/upgrade-go-deps                
branch 'floitsch/upgrade-go-deps' set up to track 'origin/floitsch/upgrade-go-deps'.
Switched to a new branch 'floitsch/upgrade-go-deps'
gaston@iMac jaguar % mkdir build
gaston@iMac jaguar % go build -o build/jag cmd/jag/main.go
gaston@iMac jaguar % build/jag version
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x16eb66a]

goroutine 1 [running]:
github.com/toitlang/jaguar/cmd/jag/commands.JagCmd.VersionCmd.func3(0xc0002a0000?, {0x1f47ac0?, 0x4?, 0x1878094?})
	/Users/gaston/Toit/debug-jaguar/jaguar/cmd/jag/commands/version.go:30 +0x4a
github.com/spf13/cobra.(*Command).execute(0xc0002a0000, {0x1f47ac0, 0x0, 0x0})
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987 +0xaa3
github.com/spf13/cobra.(*Command).ExecuteC(0xc00028e300)
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(0x1879aa7?, {0x19a3758?, 0xc000255b90?})
	/Users/gaston/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1032 +0x47
main.main()
	/Users/gaston/Toit/debug-jaguar/jaguar/cmd/jag/main.go:34 +0x190
gaston@iMac jaguar %

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -16,7 +16,7 @@ jobs:
   build_jag:
     strategy:
       matrix:
-        container: [ ubuntu-latest, ubuntu-20.04 ]
+        container: [ ubuntu-latest, ubuntu-20.04, macos-latest, windows-latest ]
 
     runs-on: ${{ matrix.container }}
     steps:
@@ -33,11 +33,13 @@ jobs:
           if [[ "${{ github.event_name }}" = "release" ]]; then
             export JAG_BUILD_RELEASE=1
           fi
-          GOOS=linux make JAG_BINARY=linux/jag jag
-          GOOS=darwin make JAG_BINARY=macos/jag jag
-          GOOS=windows make JAG_BINARY=windows/jag.exe jag
-          GOOS=linux GOARCH=arm make JAG_BINARY=linux-arm/jag jag
-          GOOS=linux GOARCH=arm64 make JAG_BINARY=linux-arm64/jag jag
+          RUNNER_OS=$(echo ${{ runner.os }} | tr '[:upper:]' '[:lower:]')
+          make JAG_BINARY=$RUNNER_OS/jag jag
+
+          if [[ $RUNNER_OS = "linux" ]]; then
+            make JAG_BINARY=$RUNNER_OS-arm/jag jag
+            make JAG_BINARY=$RUNNER_OS-arm64/jag jag
+          fi
 
       - name: Upload artifacts
         uses: actions/upload-artifact@v3

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -53,16 +53,19 @@ jobs:
             build/windows/
 
       - name: Create archive Linux
+        if: runner.os == 'Linux'
         working-directory: ./build/linux
         run: |
           tar -czf jag.tgz jag
 
       - name: Create archive Linux ARM
+        if: runner.os == 'Linux'
         working-directory: ./build/linux-arm
         run: |
           tar -czf jag-arm.tgz jag
 
       - name: Create archive Linux ARM64
+        if: runner.os == 'Linux'
         working-directory: ./build/linux-arm64
         run: |
           tar -czf jag-arm64.tgz jag

--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ JAG_TOIT_SOURCES := $(shell find src -name '*.toit') package.lock package.yaml
 JAG_GO_SOURCES := $(shell find cmd -name '*.go')
 
 # Setup Go compilation flags.
-GO_BUILD_FLAGS := GODEBUG=netdns=go
+GO_BUILD_FLAGS :=
 GO_LINK_FLAGS += -X 'main.buildDate="$(BUILD_DATE)"'
 ifdef JAG_BUILD_RELEASE
 GO_LINK_FLAGS += -X 'main.buildMode=release'
@@ -46,7 +46,7 @@ clean:
 jag: $(BUILD_DIR)/$(JAG_BINARY)
 
 $(BUILD_DIR)/$(JAG_BINARY): $(JAG_GO_SOURCES)
-	$(GO_BUILD_FLAGS) go build -tags 'netgo osusergo' -ldflags "$(GO_LINK_FLAGS)" -o $@ ./cmd/jag
+	$(GO_BUILD_FLAGS) go build -ldflags "$(GO_LINK_FLAGS)" -o $@ ./cmd/jag
 
 #############################
 # Rules for the Jaguar assets

